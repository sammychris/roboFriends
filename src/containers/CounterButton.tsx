import React, { useState } from "react";

const CounterButton = () => {
  const [count, setCount] = useState(0);
  return (
    <div>
      <button id="counter" onClick={() => setCount(count + 1)}>
        start counting
      </button>
      <div id="show-count">{count}</div>
    </div>
  );
};

export default CounterButton;
