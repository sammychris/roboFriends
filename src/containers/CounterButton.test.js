import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

const wrapper = shallow(<CounterButton />);

it("test a stateful component using snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("test a stateful component", () => {
  wrapper.find("#counter").simulate("click");
  wrapper.find("#counter").simulate("click");
  expect(wrapper.find("#show-count").text()).toEqual("2");
});
