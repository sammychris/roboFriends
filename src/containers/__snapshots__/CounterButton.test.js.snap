// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test a stateful component using snapshot 1`] = `ShallowWrapper {}`;
